src/CMakeLists.txt

```diff
@@ -79,30 +83,41 @@ target_include_directories(${PROJECT_NAME} PRIVATE ${EIGEN3_INCLUDE_DIRS})
 target_link_libraries(${PROJECT_NAME} PUBLIC Eigen3::Eigen)  # Header only library
 
 # ZLIB
-if(DEFINED ZLIB_INCLUDE_DIRS AND DEFINED ZLIB_LIBRARIES_DIRS AND DEFINED ZLIB_LIBRARIES)
-    message(STATUS "Using specific Zlib")
+if(BUILD_SUPERBUILD AND NOT USE_SYSTEM_ZLIB)
```

Comment:
Sometimes CMake says that it has found ZLIB/GTest/etc, but then some of these variables it's empty